src/Compilers/CSharp/Test/Semantic/Semantics/UnsafeTests.cs

```diff
@@ -6155,16 +6155,17 @@ class NotPointer
             CreateStandardCompilation(text, options: TestOptions.UnsafeReleaseDll).VerifyDiagnostics(
                 // (9,26): error CS0254: The right hand side of a fixed statement assignment may not be a cast expression
                 //         fixed (byte* p = (byte*)&x)
-                Diagnostic(ErrorCode.ERR_BadCastInFixed, "(byte*)&x"),
+                Diagnostic(ErrorCode.ERR_BadCastInFixed, "(byte*)&x").WithLocation(9, 26),
                 // (13,25): error CS0213: You cannot use the fixed statement to take the address of an already fixed expression
                 //         fixed (int* p = n) //CS0213 (confusing, but matches dev10)
-                Diagnostic(ErrorCode.ERR_FixedNotNeeded, "n"),
+                Diagnostic(ErrorCode.ERR_FixedNotNeeded, "n").WithLocation(13, 25),
+                // (13,25): error CS0266: Cannot implicitly convert type 'NotPointer*' to 'int*'. An explicit conversion exists (are you missing a cast?)
+                //         fixed (int* p = n) //CS0213 (confusing, but matches dev10)
+                Diagnostic(ErrorCode.ERR_NoImplicitConvCast, "n").WithArguments("NotPointer*", "int*").WithLocation(13, 25),
```

Comment:
The binder looks for something like  `fixed (int* p = &n)`  where `n` has managed value which needs pinning. It does not find that but finds that n can be converted to a pointer directly so it complains with `ERR_FixedNotNeeded`. 

From that if just goes downhill.

To continue analysis binder needs to assume something and it assumes the nearest possible valid case - like if we actually had that `&` which would result in element type being `NotPointer*` and then binder looks if element type  can convert to the pointer type that we want - `int*`, thus we have the second error.

I will see if I can just suppress that. Perhaps there is no point in analyzing element conversion when we know that `fixed` is not applicable in this case.

---
In reply to: [124925203](https://github.com/dotnet/roslyn/pull/20548#discussion_r124925203) [](ancestors = 124925203)